tests/unit/picasso/test_rest.py

```diff
@@ -35,6 +39,47 @@
 NOT_EXIST_IMG_NAME = _cfg.tests.not_exist_img_name
 NOT_EXIST_SNAP_NAME = _cfg.tests.not_exist_snap_name
 
+# The Coverage Issue for these tests was that no coverage data was
+# being generated for the server processes.
+
+# To solve this I made the server processes to run as threads as
+# coverage.py covers threads by default.
+
+# The next issue was stopping these threads when the tests were done.
+# To solve this I created a child thread that starts the server threads
+# as daemon threads and checks a global variable.
+
+# When the global variable is true the thread exits and the server threads
+# die as they are daemon threads.
+
+# This successfully results in flushing of coverage data.
+threads = []
+stop_services = False
+
+
+def start_services():
+    global threads, stop_services
+    stop_services = False
+    threads = [Thread(target=start_name_server),
+               Thread(target=start_rpc_server),
+               Thread(target=rest.start)]
+    for t in threads:
+        t.daemon = True
+        t.start()
+
+    while not stop_services:
+        time.sleep(0)
```

Comment:
curious about `time.sleep(0)` versus `pass`, or any other choice of no-op statement. not sure if it really matters...